server/src/main/scala/actors/PartitionedPortStateActor.scala

```diff
@@ -0,0 +1,210 @@
+                                val now: () => SDateLike,
+                                val terminals: List[Terminal],
+                                val journalType: StreamingJournalLike) extends Actor with ProdPartitionedPortStateActor {
+  val log: Logger = LoggerFactory.getLogger(getClass)
+
+  implicit val ec: ExecutionContextExecutor = context.dispatcher
+  implicit val mat: ActorMaterializer = ActorMaterializer.create(context)
+  implicit val timeout: Timeout = new Timeout(60 seconds)
+
+  def processMessage: Receive = {
+    case msg: SetCrunchQueueActor =>
+      log.info(s"Received crunchSourceActor")
+      flightsActor ! msg
+
+    case msg: SetSimulationActor =>
+      log.info(s"Received simulationSourceActor")
+      queuesActor ! msg
+
+    case StreamInitialized => sender() ! Ack
+
+    case StreamCompleted => log.info(s"Stream completed")
+
+    case StreamFailure(t) => log.error(s"Stream failed", t)
+
+    case flightsWithSplits: FlightsWithSplitsDiff =>
+      val replyTo = sender()
+      askThenAck(flightsWithSplits, replyTo, flightsActor)
+
+    case noUpdates: PortStateMinutes[_, _] if noUpdates.isEmpty =>
+      sender() ! Ack
+
+    case someQueueUpdates: PortStateQueueMinutes =>
+      val replyTo = sender()
+      askThenAck(someQueueUpdates.asContainer, replyTo, queuesActor)
+
+    case someStaffUpdates: PortStateStaffMinutes =>
+      val replyTo = sender()
+      askThenAck(someStaffUpdates.asContainer, replyTo, staffActor)
+
+    case GetState =>
+      log.warn("Ignoring GetState request (for entire state)")
+
+    case PointInTimeQuery(millis, request: GetStateForDateRange) =>
+      replyWithPortState(sender(), PointInTimeQuery(millis, request))
+
+    case PointInTimeQuery(millis, request: GetStateForTerminalDateRange) =>
+      replyWithPortState(sender(), PointInTimeQuery(millis, request))
+
+    case PointInTimeQuery(pitMillis, GetFlights(from, to)) =>
+      flightsActor.ask(PointInTimeQuery(pitMillis, GetStateForDateRange(from, to))).pipeTo(sender())
+
+    case PointInTimeQuery(pitMillis, GetFlightsForTerminal(from, to, terminal)) =>
+      flightsActor.ask(PointInTimeQuery(pitMillis, GetStateForTerminalDateRange(from, to, terminal))).pipeTo(sender())
+
+    case request: GetStateForDateRange =>
+      replyWithPortState(sender(), request)
+
+    case request: GetStateForTerminalDateRange =>
+      replyWithPortState(sender(), request)
+
+    case GetUpdatesSince(since, start, end) =>
+      replyWithUpdates(since, start, end, sender())
+
+    case GetFlights(from, to) =>
+      flightsActor.ask(GetStateForDateRange(from, to)).pipeTo(sender())
+
+    case GetFlightsForTerminal(from, to, terminal) =>
+      flightsActor.ask(GetStateForTerminalDateRange(from, to, terminal)).pipeTo(sender())
+  }
+
+  override def receive: Receive = processMessage orElse {
+    case unexpected => log.warn(s"Got unexpected: ${unexpected.getClass}")
+  }
+
+  def replyWithUpdates(since: MillisSinceEpoch,
```

Comment:
some of these functions might be better broken out of here so they can be tested in isolation.

Reply:
noted